lib/rational/src/Rational.cpp

```diff
@@ -60,83 +67,78 @@ Rational& Rational::operator-=(const Rational other) {
 
 Rational& Rational::operator*=(const Rational other) {
 	Rational temp(
-		this->numerator * other.numerator,
-		this->denominator * other.denominator
+		numerator_ * other.numerator_,
+		denominator_ * other.denominator_
 	);
 	temp.Reduce();
-	this->numerator = temp.numerator;
-	this->denominator = temp.denominator;
+	numerator_ = temp.numerator_;
+	denominator_ = temp.denominator_;
 	return *this;
 }
 
 Rational& Rational::operator/=(const Rational other) {
 	Rational temp(
-		this->numerator * other.denominator,
-		this->denominator * other.numerator
+		numerator_ * other.denominator_,
+		denominator_ * other.numerator_
 	);
 	temp.Reduce();
-	this->numerator = temp.numerator;
-	this->denominator = temp.denominator;
+	numerator_ = temp.numerator_;
+	denominator_ = temp.denominator_;
 	return *this;
 }
 
 double Rational::ToDouble() {
-	return static_cast<double>(numerator) /
-		static_cast<double>(denominator);
+	return double(numerator_) /
+		denominator_;
 }
 
 std::ostream& operator<<(std::ostream& os, const Rational other) {
 	return other.Print(os);
 }
 
 bool operator==(const Rational lhs, const Rational rhs) {
-	Rational a = lhs.Reduce();
-	Rational b = rhs.Reduce();
-
-	if (a.numerator == b.numerator &&
-		a.denominator == b.denominator)
-		return true;
-	else
-		return false;
+	return 
+		rhs.numerator_ == lhs.numerator_ &&
+		rhs.denominator_ == lhs.denominator_;
 }
 
 bool operator!=(const Rational lhs, const Rational rhs) {
 	return !operator==(lhs, rhs);
 }
 
 auto Rational::Print(std::ostream& os) const->std::ostream& {
-	os << this->numerator << "/" << this->denominator;
+	os << numerator_ << "/" << denominator_;
 	return os;
 }
 
-Rational Rational::Reduce() const {
-	int divisor = Gcd(this->numerator, this->denominator);
-	return Rational{
-		this->numerator / divisor,
-		this->denominator / divisor
-	};
+void Rational::Reduce() {
+	const auto divisor = Gcd(numerator_, denominator_);
+	Rational temp = *this;
+	temp.numerator_ /= divisor;
+	temp.denominator_ /= divisor;
+	*this = temp;
```

Comment:
It looks like we don't need to create a temp object.
You only need to change members
numerator_ /= divisor;
denominator_ /= divisor;